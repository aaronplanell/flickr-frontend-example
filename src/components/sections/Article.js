import React from 'react';

const commonStyle = {
  background: 'white',
  borderColor: 'grey',
  borderStyle: 'dashed',
  borderRadius: '0pt',
  paddingLeft: '20px'
};

const Article = ({alert}) => {
  if (alert || alert === '') {
    return (
      <article style={{...commonStyle, paddingTop: '15px', color: 'brown', fontFamily: 'courier'}}>
        {alert}
      </article>
    );
  } else {
    return (
      <article style={{...commonStyle}}>
      </article>
    );
  }
}

Article.propTypes = {
  alert: React.PropTypes.string.isRequired
}

export default Article;
