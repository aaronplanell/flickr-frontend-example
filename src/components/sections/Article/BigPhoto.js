import React from 'react';

const BigPhoto = ({photo, selectPhoto}) => (
  <div style={{ position: 'relative', zIndex: 1000,  width: '100%', height: '100%'}}>
    <img
      alt={photo.id}
      id={photo.id}
      src={photo.original}
      onClick={(event) => selectPhoto({})}
      style={{
        marginLeft: 'auto',
      	marginRight: 'auto',
      	display: 'block'
      }}
    />
  </div>
);

BigPhoto.propTypes = {
  photo: React.PropTypes.object.isRequired,
  selectPhoto: React.PropTypes.func
}

export default BigPhoto;
