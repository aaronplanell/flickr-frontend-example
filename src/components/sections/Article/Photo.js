import React from 'react';

const styleOfPhoto = {
  paddingRight: '5px',
  opacity: 0.9
}

const Photo = ({photo, selectPhoto}) => (
  <img
    alt={photo.id}
    id={photo.id}
    src={photo.source}
    style={styleOfPhoto}
    onClick={(event) => selectPhoto(photo)}
    onMouseOver={(event) => event.target.style.opacity = 1.0}
    onMouseLeave={(event) => event.target.style.opacity = 0.9}
  />
);

export default Photo;
