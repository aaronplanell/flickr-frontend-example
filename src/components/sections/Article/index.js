import React from 'react';
import Photo from './Photo';
import BigPhoto from './BigPhoto';

const commonStyle = {
  background: 'white',
  borderColor: 'grey',
  borderStyle: 'dashed',
  borderRadius: '0pt',
  paddingLeft: '20px'
};

const renderAlert = (alert) => {
  return (
    <article style={{...commonStyle, paddingTop: '15px', color: 'brown', fontFamily: 'courier'}}>
      {alert}
    </article>
  );
}

const renderListOfPhotos = (selectedPhotos, selectPhoto) => {
  return (
    <article style={{...commonStyle}}>
      { selectedPhotos.map( photo => <Photo photo={photo} key={photo.id} selectPhoto={selectPhoto}/> ) }
    </article>
  );
}

const renderBigPhoto = (selectedPhoto, selectPhoto) => {
  return (
    <BigPhoto photo={selectedPhoto} selectPhoto={selectPhoto}/>
  );
}

const Article = ({alert, selectedPhotos, currentViewSize, selectPhoto, selectedPhoto}) => {
  if (alert && alert !== '') {
    return renderAlert(alert);
  } else {
    if (selectedPhoto && selectedPhoto.hasOwnProperty('id')) {
      return renderBigPhoto(selectedPhoto, selectPhoto);
    } else {
      return renderListOfPhotos(selectedPhotos, selectPhoto);
    }
  }
}

Article.propTypes = {
  alert: React.PropTypes.string.isRequired,
  selectedPhotos: React.PropTypes.array.isRequired,
  currentViewSize: React.PropTypes.number.isRequired,
  selectPhoto: React.PropTypes.func.isRequired,
  selectedPhoto: React.PropTypes.object.isRequired
}

export default Article;
