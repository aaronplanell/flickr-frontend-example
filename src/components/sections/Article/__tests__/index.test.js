import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Article from '../';

const alert = '';
const currentViewSize = 4;
const selectedPhotos = [
  {
    "id": "32393549123",
    "source": "https://farm1.staticflickr.com/748/32393549123_7eeaf3987f_n.jpg",
    "original": "https://farm1.staticflickr.com/748/32393549123_aa850eb989_o.jpg"
  },
  {
    "id": "33080731281",
    "source": "https://farm1.staticflickr.com/766/33080731281_bc00c16a4f_n.jpg",
    "original": "https://farm1.staticflickr.com/766/33080731281_66f6a050aa_o.jpg"
  },
  {
    "id": "33208490285",
    "source": "https://farm1.staticflickr.com/711/33208490285_7a90bc6f0d_n.jpg",
    "original": "https://farm1.staticflickr.com/711/33208490285_6c6e229e8c_o.jpg"
  },
  {
    "id": "32393539423",
    "source": "https://farm4.staticflickr.com/3767/32393539423_86561428ab_n.jpg",
    "original": "https://farm4.staticflickr.com/3767/32393539423_8d5092d68b_o.jpg"
  },
  {
    "id": "32364211744",
    "source": "https://farm4.staticflickr.com/3822/32364211744_1f17268760_n.jpg",
    "original": "https://farm4.staticflickr.com/3822/32364211744_1ee859de4d_o.jpg"
  },
  {
    "id": "33208484225",
    "source": "https://farm1.staticflickr.com/597/33208484225_dae04ffe50_n.jpg",
    "original": "https://farm1.staticflickr.com/597/33208484225_dbd65b6a8d_o.jpg"
  },
  {
    "id": "33051771402",
    "source": "https://farm4.staticflickr.com/3851/33051771402_4054d32ddf_n.jpg",
    "original": "https://farm4.staticflickr.com/3851/33051771402_eda8df9a85_o.jpg"
  },
  {
    "id": "33080707611",
    "source": "https://farm1.staticflickr.com/630/33080707611_a8ec266c8b_n.jpg",
    "original": "https://farm1.staticflickr.com/630/33080707611_7d02d99455_o.jpg"
  },
  {
    "id": "33208476055",
    "source": "https://farm1.staticflickr.com/599/33208476055_84361a4d89_n.jpg",
    "original": "https://farm1.staticflickr.com/599/33208476055_194ff6f531_o.jpg"
  }
];

const selectedPhoto = {
  id: 1234,
  original: 'http://host:port/image.jpg'
}

const selectPhoto = jest.fn();

it('Photo Renders OK', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Article
      alert={alert}
      currentViewSize={currentViewSize}
      selectedPhotos={selectedPhotos}
      selectPhoto={selectPhoto}
      selectedPhoto={selectedPhoto}
    />,
    div);
});

it('Photo Snapshot matchs', () => {
  const tree = renderer.create(
    <Article
      alert={alert}
      currentViewSize={currentViewSize}
      selectedPhotos={selectedPhotos}
      selectPhoto={selectPhoto}
      selectedPhoto={selectedPhoto}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
