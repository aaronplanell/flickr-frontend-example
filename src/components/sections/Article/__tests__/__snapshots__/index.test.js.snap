exports[`test Photo Snapshot matchs 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "position": "relative",
      "width": "100%",
      "zIndex": 1000,
    }
  }>
  <img
    alt={1234}
    id={1234}
    onClick={[Function]}
    src="http://host:port/image.jpg"
    style={
      Object {
        "display": "block",
        "marginLeft": "auto",
        "marginRight": "auto",
      }
    } />
</div>
`;
